### DISCUSSION TITLE
NWK conflict

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1275] [Q&A]
We have had a strange behaviour on a Ezsp - 6.10.3.0 build 297

> 2023-11-02 05:15:01,344 WARNING :NWK conflict is reported for 0x7b29
> 2023-11-02 05:15:01,347 WARNING :Found ec:1b:bd:ff:fe:65:ac:ca device for 0x7b29 NWK conflict: None None
> 2023-11-02 05:15:01,355 INFO    :b'handle_leave (0x7b29 ec:1b:bd:ff:fe:65:ac:ca)'

> 2023-11-02 10:38:45,376 WARNING :NWK conflict is reported for 0xa080
> 2023-11-02 10:38:45,377 WARNING :Found 84:fd:27:ff:fe:5a:da:4e device for 0xa080 NWK conflict: None None
> 2023-11-02 10:38:45,381 INFO    :b'handle_leave (0xa080 84:fd:27:ff:fe:5a:da:4e)'

we are using the 0.59.0 zigpy module and 0.36.8 bellows

**Comment by pipiche38** [comment]
Investigating further in the log, this looks happening time to time, but usually there is a device announcement coming , while here nothing happen !

**Reply by MattWestb** [reply]
From the MAC is one Silabs chipped devices and they was having one bug in the Zigbee stack that was doing that the stack was thinking it was one conflict and changing its NVK and most IKEA devices is doing it some time but not oft. I think it can also being triggered in the NCP side and not only by the device its self.

**Reply by pipiche38** [reply]
do you think it can be linked to a zigpy or bellows code changes ?

**Reply by MattWestb** [reply]
No its in the Zigbee stack (network layer) in the NCP / RCP (Zigbeed) and / or in the device that is complaining and not in the application layer.
I think the coordinator side shall working better if running the latest NCP / RCP firmware (7 / 4.3.2) that have many bug fixed but only if its the coordinator side its not helping if the device is doing the problem.

**Reply by pipiche38** [reply]
```
2023/11/03 13:25:02 Elelabs_EzspFwUtility:   Elelabs Zigbee adapter detected:
2023/11/03 13:25:02 Elelabs_EzspFwUtility:   Adapter: ELR023
2023/11/03 13:25:02 Elelabs_EzspFwUtility:   Firmware: 6.10.3-41
2023/11/03 13:25:02 Elelabs_EzspFwUtility:   EZSP v8
```

But it looks like Elelabs doesn't provide any firmware upgrade. So I'm afraid to be stuck with that.

**Reply by MattWestb** [reply]
The EZSP 6.10.3 (and the newer bug fixed of it) is still the recommended version for **ALL** production system.
Its have some bugs like this one but but not new bugs that the 7.X have implanting.
I think the hardware config for the ELR023 is out so it shall not being any problem making one update NCP firmware for it also one RCP is some on need it for Thread and Zigbee.

**Reply by MattWestb** [reply]
Here you have the hardware settings used for your firmware but they have only old ones but can making updated then having the right hardware setting for it.
https://github.com/zha-ng/EZSP-Firmware#elelabs-elu013-and-elr023

**Comment by Hedda** [comment]
Suggest also report issue to Elelabs -> https://github.com/Elelabs/elelabs-zigbee-ezsp-utility/issues


>> I think the coordinator side shall working better if running the latest NCP / RCP firmware (7 / 4.3.2) that have many bug fixed but only if its the coordinator side its not helping if the device is doing the problem.
>
> But it looks like Elelabs doesn't provide any firmware upgrade. So I'm afraid to be stuck with that.

FYI, while true that Elelabs do not provide newer firmware, there are several third-party community developers providing unofficial firmware. Disclaimer is of course to use at your own risk but these have at least so far been proven to be trusted sources within the community. See ex:

https://github.com/darkxst/silabs-firmware-builder -> check out darkxst's "manifestjs-SL" branch which contains cutting-edge releases-> https://github.com/darkxst/silabs-firmware-builder/tree/manifestjs-SL/firmware_builds/elelabs-elx0x3

You can also built the firmware yourself using that silabs-firmware-builder to get whatever version you want:

https://github.com/darkxst/silabs-firmware-builder/blob/main/README.md#building-locally

(more discussions about Elelabs firmware in https://github.com/darkxst/silabs-firmware-builder/issues/8 and https://github.com/ksjh/silabs-firmware-builder/issues/17 )

Patch forked from giannello/'s "feature/elelabs-support" branch ->
https://github.com/giannello/silabs-firmware-builder/tree/feature/elelabs-support

PS: Newer official firmware requests has been sent to Elelabs but they seem to have gone missing or is at least no longer posting.
